### PR TITLE
MRG: Allow silencing of "The unit for channel(s) ... has changed" warnings

### PR DESCRIPTION
**Pull request by hoechenberger**
In MNE-BIDS we commonly change the units during an ordinary reading operation. This is expected behavior and should not yield a warning.

However, trying to **catch and filter** the warning that MNE emits is quite cumbersome, esp. inside the MNE-BIDS test suite.

I, therefore, believe the best approach is to allow the silencing of the warnings in MNE-Python directly.

This PR adds a new parameter, `on_unit_change`, to the `.set_channel_types()` methods. The default is to warn, but users can now also choose to `'ignore'` this.

x-ref https://github.com/mne-tools/mne-bids/issues/1100

cc @sappelhoff

### CODE REVIEW

**Comment by hoechenberger** [comment]
CI failures seem unrelated

**Comment by hoechenberger** [comment]
Note that in the few tests where I removed the `pytest.warns()` context managers, I made sure those tests were not designed to actually test for those warnings – the warnings simply had to be caught to prevent the tests of some other functionality from failing. The tests that **do** focus on the warnings still remain in the test suite.

File: mne/channels/channels.py [hunk 1]
```diff
@@ -308,14 +308,20 @@ def _set_channel_positions(self, pos, names):
...
 
         Parameters
         ----------
         mapping : dict
-            A dictionary mapping a channel to a sensor type (str), e.g.,
+            A dictionary mapping channel names to sensor types, e.g.,
             ``{'EEG061': 'eog'}``.
+        on_unit_change : 'warn' | 'ignore'
+            Whether to emit a warning (default) if the measurement unit of a
+            channels is changed automatically to match the new sensor
```
Comment (hoechenberger) [inline mne/channels/channels.py:321]:
```suggestion
            channel is changed automatically to match the new sensor
```

File: mne/channels/channels.py [hunk 2]
```diff
@@ -337,6 +343,10 @@ def set_channel_types(self, mapping, verbose=None):
 
         .. versionadded:: 0.9.0
         """
+        _check_option(
+            'on_unit_change', value=on_unit_change,
+            allowed_values=('warn', 'ignore')
+        )
```
Comment (larsoner) [inline mne/channels/channels.py:349]: I don't see a reason not to allow `raise`. And if we do that we might as well use our standard `on_*` helpers. This should make the code more DRY

**Review by larsoner — CHANGES_REQUESTED** [review]

**Comment by hoechenberger** [comment]
Style Check error seems unrelated

**Comment by hoechenberger** [comment]
I believe the CI errors are unrelated

**Comment by larsoner** [comment]
Thanks @hoechenberger !

**Pull request merged**